tests/http_server_test.cc

```diff
@@ -1043,3 +1051,40 @@ TEST(http_server_test, server_with_content_encoding_deflate)
     ASSERT_EQ(originalUncompressedData, newlyDecompressedData);
 }
 #endif
+
+TEST(http_server_test, http_server_is_not_leaked)
+{
+    const auto number_of_fds = [] {
+        using filesystem::directory_iterator;
+        const filesystem::path fds_dir { "/proc/self/fd" };
+
+        if (!filesystem::exists(fds_dir))
+        {
+            return directory_iterator::difference_type(0);
+        }
+
+        return std::distance(directory_iterator(fds_dir), directory_iterator {});
+    };
+
+    const auto fds_before = number_of_fds();
+    const Pistache::Address address("localhost", Pistache::Port(0));
+
+    auto server      = std::make_unique<Http::Endpoint>(address);
+    auto flags       = Tcp::Options::ReuseAddr;
+    auto server_opts = Http::Endpoint::options().flags(flags).threads(4);
+    server->init(server_opts);
+    server->setHandler(Http::make_handler<PingHandler>());
+    server->serveThreaded();
+    server->shutdown();
+    server.reset();
+
+    if (fds_before > 0)
+    {
+        const auto fds_after = number_of_fds();
+        ASSERT_EQ(fds_before, fds_after);
+    }
+    else
+    {
+        std::cout << "NOTE: Please use Valgrind with '--track-fds=yes' option for this test" << std::endl;
```

Comment:
@tyler92, I am a bit concerned about this. Does the user need to run the unit test in valgrind(1) in order to verify that it passed? The reason I ask is it would be cumbersome for a user to have to manually do that, as opposed to the build environment automatically doing that. If it can't do that, then if the test fails we'll never know during CI.

Reply:
I agree with you. It's related to the first question from [another thread](https://github.com/pistacheio/pistache/pull/1158#discussion_r1378264402). The logic here is the following - we need to handle the case when our "leak detector" fails. In this case, we can do one of the following: fail the test, pass the test, or pass with a warning. I chose the last option, but if we have full confidence in this leak detection approach, we may fail the test.

Reply:
@kiplingw Probably there is a misunderstanding. When I said "leak detection approach", I meant not valgrind, but checking file descriptors count. Now I suggest removing anything about Valgrind in my test, and it will look like this:

```cpp
TEST(http_server_test, http_server_is_not_leaked)
{
    const auto number_of_fds = [] {
        using filesystem::directory_iterator;
        const filesystem::path fds_dir { "/proc/self/fd" };

        if (!filesystem::exists(fds_dir))
        {
            return directory_iterator::difference_type(0);
        }

        return std::distance(directory_iterator(fds_dir), directory_iterator {});
    };

    const auto fds_before = number_of_fds();
    ASSERT_GT(fds_before, 0);

    // ...

    const auto fds_after = number_of_fds();
    ASSERT_EQ(fds_before, fds_after);
```

Valgrind was a fallback option in my original idea, but it has downsides.

Reply:
@dennisjenkins75, thoughts?

Reply:
Since you probably intend to remove the Valgrind fallback this isn't very relevant, but in general when skipping tests in GoogleTest you should use `GTEST_SKIP()` (documented at <https://google.github.io/googletest/advanced.html#skipping-test-execution>) since it will properly signal to the testing environment that a test case was skipped.